analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.cs

```diff
@@ -218,13 +219,87 @@ public void Branching_ConstraintTrackedSeparatelyInBranches()
     value = false;
 }
 Tag(""End"", value);";
-            var validator = SETestContext.CreateCS(code).Validator;
-            validator.ValidateExitReachCount(2);    // Once with True constraint, once with False constraint on "value"
+            var validator = SETestContext.CreateCS(code, new PreserveTestCheck("value")).Validator;
+            validator.ValidateExitReachCount(2); // Once with True constraint, once with False constraint on "value"
             validator.TagValues("End").Should().HaveCount(2)
                 .And.ContainSingle(x => x.HasConstraint(BoolConstraint.True))
                 .And.ContainSingle(x => x.HasConstraint(BoolConstraint.False));
         }
 
+        [TestMethod]
+        public void Branching_FieldSymbolsAreNotCleaned()
+        {
+            const string code = @"
+if (boolParameter)
+{
+    field = 1;
+}
+else
+{
+    field = 2;
+}
+Tag(""End"", field);";
+
+            var postProcess = new PostProcessTestCheck(x =>
+            {
+                if (x.Operation.Instance.Kind == OperationKind.Literal && x.Operation.Instance.ConstantValue.Value is int value)
+                {
+                    if (value == 1)
+                    {
+                        return x.SetOperationConstraint(TestConstraint.First);
+                    }
+                    else if (value == 2)
+                    {
+                       return x.SetOperationConstraint(TestConstraint.Second);
+                    }
+                }
+                return x.State;
+            });
+            var validator = SETestContext.CreateCS(code, postProcess).Validator;
+            validator.ValidateExitReachCount(2);    // Once with First constraint, once with Second constraint on "value"
+            validator.TagValues("End").Should().HaveCount(2)
+                .And.ContainSingle(x => x.HasConstraint(TestConstraint.First))
+                .And.ContainSingle(x => x.HasConstraint(TestConstraint.Second));
```

Comment:
This assertion doesn't do much, because the value is preserved by the `Tag` itself. 
So we don't really need two complicated states.
Having that, we can do this
```
if (boolParameter)
{
    field = 42;
}
```
to simplify the check just to
```
new PostProcess(OperationKind.Literal, x => x.SetOperationConstraint(TestConstraint.Dummy))
```
And then, we should preserve the `ValidateExitReachCount` with the comment above, because that's the important thing. And we can remove this assertion.
```suggestion
```